ci_tools/check_pylint_commands.py

```diff
@@ -20,6 +21,40 @@
                             re.compile('./tests/semantic/scripts/expressions.py'):['unused-variable'],
                             re.compile('./tests/semantic/scripts/calls.py'):['unused-variable']}
 
+def run_pylint(file, flag, messages):
+    """
+    Check for an expected pylint disable flag. If the flag is present
+    then it is ignored by removing it from the list. Otherwise if the
+    file raises the error a message is saved recommending that the
+    flag be disabled in the file.
+
+    Parameters
+    ----------
+    file : file object
+        The file being analysed.
+    flag : str
+        The name of the flag being investigated.
+    messages : list
+        The list of messages which should be printed.
+    """
+    with subprocess.Popen([shutil.which('pylint'), file, '--disable=all', f'--enable={flag}']) as r:
+        r.communicate()
+        result = r.returncode
+    if result:
+        output_item = {
+            "title":f"[Warning]: Flag needs to be disabled",
+            "summary":f"Feel free to disable {flag} in {file}",
+            "annotations": []
+        }
+        output_item["annotations"].append({
+            "annotation_level":"Warning",
+            "start_line":1,
+            "end_line":1,
+            "path":file,
+            "message":f"disable: {flag}"
+        })
+        messages["output"].append(output_item)
```

Comment:
I think there has been a little confusion here. The title and summary should be for the entire run not for each error. So here you should simply have:

```suggestion
        output_item["annotations"].append({
            "title":f"[Warning]: Flag needs to be disabled",
            "annotation_level":"Warning",
            "start_line":1,
            "end_line":1,
            "path":file,
            "message":f"Feel free to disable {flag} in this file"
        })
```